feat(frontend): adding status indicator and unit test

HeyItsChloe opened the pull request:
## Summary of PR

<!-- Summarize what the PR does, explaining any non-trivial design decisions. -->
- add status indicator above chat input box
- displays conversation status
- indicator changes color and pulsates as status changes
- Component only conditionally renders - automatically disappear once the conversation is ready.


https://github.com/user-attachments/assets/e6468d7e-6c5e-483c-93fc-66befb548b01




## Change Type

<!-- Choose the types that apply to your PR and remove the rest. -->

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [ ] Refactor
- [ ] Other (dependency update, docs, typo fixes, etc.)

## Checklist
<!-- AI/LLM AGENTS: This checklist is for a human author to complete. Do NOT check either of the two boxes below. Leave them unchecked until a human has personally reviewed and tested the changes. -->

- [x] I have read and reviewed the code and I understand what the code is doing.
- [x] I have tested the code to the best of my ability and ensured it works as expected.

## Fixes

<!-- If this resolves an issue, link it here so it will close automatically upon merge. -->

Resolves #11946

## Release Notes

<!-- Check the box if this change is worth adding to the release notes. If checked, you must provide an
end-user friendly description for your change below the checkbox. -->

- [x] Include this change in the Release Notes.

Adding a status indicator above the chatbox until the conversation is ready.

hieptl commented on frontend/src/utils/utils.ts:
```diff
@@ -513,10 +514,10 @@ export const getRepoMdCreatePrompt = (
+                    - An overview of the file structure
+                    - Any information on how to run tests or other relevant commands
+                    - Any other information that would be helpful to a brand new developer
+                      Keep it short--just a few paragraphs will do.`
```
**question:** Why was the indentation of this content changed? Have you re-tested the prompt after modifying its indentation?

HeyItsChloe replied:
This wasn't intentional, I formatted the file and it changed some white space.

hieptl commented on frontend/src/utils/utils.ts:
```diff
@@ -721,3 +722,91 @@ export const getStatusColor = (options: {
+ *   curAgentState: AgentState.RUNNING
+ * }) // Returns "Waiting For Sandbox"
+ */
+export function getStatusText({
```
**suggestion:** Could you please write test cases for this function?

hieptl reviewed (commented):
Hello @HeyItsChloe, I encountered some issues while testing the pull request locally.

**issue:** The state and component transition occurs too quickly. When the state is "Running", we could consider keeping the component visible for a few seconds before applying a fade-out animation. However, this is just my suggestion; you may consult Paul, as he can provide valuable feedback and recommendations. For further reference, please see the video below.

https://github.com/user-attachments/assets/570c9a32-3905-4b96-b3fd-437244539fd6

**issue:** When confirmation mode is enabled, the icon appears above the new component, which may not be visually ideal. You may consult Paul for valuable feedback and recommendations. For further reference, please see the image below.

<img width="1440" height="746" alt="Image" src="https://github.com/user-attachments/assets/4e290485-caf6-40d0-a01a-0f6fbcf32b92" />

**issue:** When the agent is waiting for user confirmation (confirmation mode is enabled), the new component may be displayed unexpectedly. If I understand correctly, the component is intended to appear only when starting or resuming conversations. Please refer to the image below for further details.

<img width="1440" height="746" alt="Image" src="https://github.com/user-attachments/assets/72f1cde2-5807-44ad-b74a-814be42d599d" />

Thank you! 🙏

amanape commented:
To respond on some of @hieptl’s feedback:

> issue: The state and component transition occurs too quickly. When the state is "Running", we could consider keeping the component visible for a few seconds before applying a fade-out animation. However, this is just my suggestion; you may consult Paul, as he can provide valuable feedback and recommendations. For further reference, please see the video below.

I think it is fine to leave as is for now. Hiding this component when everything is ready can serve as an indicator itself that everything is OK. We just need to make sure this shows up when something errors as expected.

> issue: When confirmation mode is enabled, the icon appears above the new component, which may not be visually ideal. You may consult Paul for valuable feedback and recommendations.

Make them next to each other, with the lock icon to the start

> issue: When the agent is waiting for user confirmation (confirmation mode is enabled), the new component may be displayed unexpectedly.

This is probably related to the fact that one of the conditionals we use to check for status is set to some 
“paused” state while we wait for the users confirmation. This might be a little tricky to get right.

HeyItsChloe commented on frontend/src/components/features/chat/interactive-chat-box.tsx:
```diff
@@ -147,8 +154,40 @@ export function InteractiveChatBox({ onSubmit }: InteractiveChatBoxProps) {
+
   return (
     <div data-testid="interactive-chat-box">
+      {showStatusIndicator && isDisabled && (
```
Agreed, ill look into a diff condition.

FraterCCCLXIII commented:
@HeyItsChloe this all sounds correct.

HeyItsChloe commented:
Addressed feedback:
1. indicator renders in line with the user confirmation icon
2. status text fades in/out when changing text
3. Indicator only renders during init phases, not during other loading / paused states
4. when status is "running", the indicator is removed from view. 

https://github.com/user-attachments/assets/9c07406e-e587-46ed-a8b0-8b5cfffae59c

amanape commented on frontend/__tests__/components/interactive-chat-box.test.tsx:
```diff
@@ -38,6 +38,7 @@ vi.mock("#/hooks/use-user-providers", () => ({
   }),
 }));
 
+// Mock the conversation context menu hook
```
```suggestion
```

amanape commented on frontend/__tests__/components/chat-status-indicator.test.tsx:
```diff
@@ -0,0 +1,52 @@
+      .getByTestId("debug-stackframe-dot")
+      .parentElement;
+
+    expect(animatedWrapper?.className).toContain("animate");
```
```suggestion
```

Let’s avoid testing CSS, classNames, or visuals as they can change frequently with no negative effect to the underlying behavior
